
syntax = "proto3";

package task;

message Empty {}

message Task {
    string id = 1;
    string configuration_id = 2;
    string location_id = 3;
    string user_id = 4;
    string role_id = 5;
    int64 due_date = 6;
    bool completed = 7;
}

message CreateTaskRequest {
    string configuration_id = 1;
    string location_id = 2;
    string user_id = 3;
    string role_id = 4;
    int64 due_date = 5;
}

message CompleteTaskRequest {
    string task_id = 1;
}

message DeleteTaskRequest {
    string task_id = 1;
}

message GetTaskRequest {
    string task_id = 1;
}

message GetAllTasksRequest {
    int32 page = 1;
    int32 limit = 2;
}

message GetAllTasksResponse {
    repeated Task tasks = 1;
}

service TaskService {
    rpc CreateTask(CreateTaskRequest) returns (Task);
    rpc CompleteTask(CompleteTaskRequest) returns (Task);
    rpc DeleteTask(DeleteTaskRequest) returns (Empty);
    rpc GetTask(GetTaskRequest) returns (Task);
    rpc GetAllTasks(GetAllTasksRequest) returns (GetAllTasksResponse);
}
